### PR TITLE
chore(aws provider): Switch to native-tls

### PR DESCRIPTION
**Pull request by jszwedko**
The bulk of our dependencies use OpenSSL so we prefer it when possible to reduce variance. We may
also consider [FIPS compliance](https://github.com/vectordotdev/vector/issues/8435) in the future;
which rustls does not have support for.

Closes: #15905

Signed-off-by: Jesse Szwedko <jesse.szwedko@datadoghq.com>

### CODE REVIEW

**Review by spencergilbert — APPROVED** [review]
yolo, did you check to see if rustls was still in our dep tree after this change?

**Comment by spencergilbert** [comment]
Oh we also wanted to mention this in the upgrade guide, no?

**Comment by jszwedko** [comment]
> Oh we also wanted to mention this in the upgrade guide, no?

I had just planned to include it in the changelog since it isn't a breaking change or deprecation, but I'm remembering we did start a pattern of including potentially impactful changes in 0.27.0 so I'll add it.

**Comment by jszwedko** [comment]
Added upgrade note in [2f955d9](https://github.com/vectordotdev/vector/pull/16335/commits/2f955d9a8cd257c11fc10377928fb68545fd22e9)

**Review by davidhuie-dd — APPROVED** [review]

File: website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md [hunk 1]
```diff
@@ -23,3 +27,18 @@ We cover them below to help you upgrade quickly:
 
 The `units` option had been replaced by `include_units` in the `0.10` release. This release marks the final removal
 of this option, please update your configurations to use the `include_units` option instead.
+
+### Potentially impactful changes
+
+#### AWS components now use OpenSSL as the TLS implementation {#aws-openssl}
+
+In this release, Vector's AWS integrations had their TLS implementation swapped from
+[`rustls`][rusttls] to OpenSSL. We don't expect any user visible impacte,
```
Comment (spencergilbert) [inline website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md:36]:
```suggestion
[`rustls`][rusttls] to OpenSSL. We don't expect any user visible impact,
```

**Review by spencergilbert — CHANGES_REQUESTED** [review]

File: website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md (hunk 1 above)
Comment (spencergilbert) [reply website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md:36]: Unless this is fancy spelling I'm not aware of 😉

Comment (jszwedko) [reply website/content/en/highlights/2023-02-28-0-28-0-upgrade-guide.md]: Need `check-spelle` 😂

**Review by spencergilbert — APPROVED** [review]

**Comment by jszwedko** [comment]
Merging since Windows build failure existed on `master`.

**Pull request merged**